common/src/Microsoft.Azure.IIoT.Storage.CosmosDb/src/Services/DocumentQuery.cs

```diff
@@ -60,7 +60,8 @@ internal sealed class DocumentQuery : ISqlClient {
 
                     EnableCrossPartitionQuery = pk == null
                 });
-            return new DocumentInfoFeed<T>(query.AsDocumentQuery(), _logger);
+            _query = query.AsDocumentQuery();
```

Comment:
`DocumentQuery` object is not tied to any specific query on initialization, which means/implies that it can be used to retrieve elements for different queries. If that is done, then those different queries will interfere with each other and `Continue()` for one of them may return results for the other query. And with the current interface of the class there is nothing in place to stop that scenario from happening. I think that we have a few options here:

* couple `DocumentQuery` object to a specific query
* query string should be passed to `Continue()` method and corresponding `IDocumentQuery<Document>` object should be recreated
* query string should be passed to `Continue()` method and corresponding `IDocumentQuery<Document>` object can be stored and retrieved from an internal cache that would tie it with the original query string

The other problem that I see with storing `_query` is that its continuation token can be used only once. So if there are many matching elements in the database, then 2nd call to `Continue()` bellow will most probably still get the results for 1st call as the continuation token has not been updated. The fact that `Continue()` bellow doesn't actually use passed `continuationToken` also points in that direction, unless I am missing bigger picture here.

Reply:
@cristipogacean, one more point, `JobDatabase.QueryAsync()` is called from several places. One path is `WorkersController.GetAvailableJobAsync()` -> `DefaultJobOrchestrator.GetAvailableJobAsync()`, and the others are through `JobsController` -> `DefaultJobService` via `ListJobsAsync()` and `QueryJobsAsync()` methods. And in case of services-all they are all going to use the same instance of `JobDatabase`. So we should make it and the client that it uses stateless as there will be concurrent calls from those API controllers to the `JobDatabase` with different queries.

Reply:
I believe the `Continue()` logic continuation token logic is designed to be executed on the very same _query object without the need to interfere with it's internal continuationToken state. The actual implementation was build on different assumptions.

the actual implementation works just fine with multiple calls with continuation tokens on the same query.

Reply:
@karok2m, yes, you're right, I switched back to stateless.